### ISSUE TITLE
Accessibility, 508 Compliance: List of APIs/Products (Dropdown) Widgets Keyboard Functionality and Button Name Attribute

### ISSUE THREAD

**Issue by jdm4344** [Azure/api-management-developer-portal#1551]
## Bug description

The List of APIs (Dropdown) and List of Products (Dropdown) widgets can not be minimized with keyboard controls, i.e. Enter or Spacebar keys when they are the object of focus on a page. The button element on both widgets is also missing an accessibility name attribute. 

### Functionality
1. Go to a page with either a List of APIs (Dropdown) or List of Products (Dropdown) widget
2. Use the Tab key to move keyboard focus to the button element of the widget
3. Use the Enter key or Spacebar key to expand the widget, widget expands.
4. Attempt to use the Enter or Spacebar key to minimize. Widget does not minimize, can only be done via mouse interaction by clicking on another part of the web page

### Accessibility Attributes
Without tool:

1. Go to a page with either a List of APIs (Dropdown) or List of Products (Dropdown) widget
2. Right click on the List of APIs (Dropdown) or List of Products (Dropdown) widget button element and choose Inspect option
3. Verify via the inspector that no accessible name, innerText, or [title] attribute is present. 

Using the [SSA ANDI Tool](https://www.ssa.gov/accessibility/andi/help/install.html):
1. Navigate to a page with either a List of APIs (Dropdown) or List of Products (Dropdown) widget
2. Open the ANDI tool in the browser
3. Click the Try It button on the widget to open the panel
4. Click ANDI's refresh button (top-right corner, leftmost button)
5. Accessibility alert appears in ANDI "Elements with No Accessible Name - Button has no accessible name, innerText, or [title]"

## Expected behavior

The List of APIs (Dropdown) and List of Products (Dropdown) widgets can be expanded and minimized using keyboard-only interaction. The widget's button element has an accessible name, associated <label> element, or [title] attribute that makes it identifiable/recognizable by accessibility software. Note: this is different from the aria-expanded attribute that is currently present on the element.

## Is your portal managed or self-hosted?

Managed

## Release tag or commit SHA (if using self-hosted version)

2.13.0

## Environment

 - Operating system: MacOS Monterey 12.0.1 - Macbook Pro 15-inch, 2017 model
 - Browser: Google Chrome
 - Version: 98

### Functionality:
Related Section 508 Trusted Tester step: https://section508coordinators.github.io/TrustedTester/keyboard.html#check-211-keyboard-access
Related WCAG success criteria: https://www.w3.org/TR/UNDERSTANDING-WCAG20/keyboard-operation-keyboard-operable.html
### Accessibility Attributes
Related Section 508 Trusted Tester step: https://section508coordinators.github.io/TrustedTester/forms.html#check-131-programmatic-label
Related WCAG success criteria: https://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-programmatic.html

<img width="1677" alt="Screen Shot 2022-02-14 at 10 19 38 AM" src="https://user-images.githubusercontent.com/31706474/153894818-15e02354-b487-4306-afee-2e77d0202561.png">

<img width="1024" alt="Screen Shot 2022-02-14 at 10 44 08 AM" src="https://user-images.githubusercontent.com/31706474/153897024-7bd567ec-2466-4027-94d2-f026f7ed4b47.png">

**Comment by jdm4344** [comment]
Updated issue as of release 2.13.0. The widgets can now be expanded with keyboard controls, but still cannot be minimized. Added details on missing name attribute on the widget's button element, including screenshots.

**Comment by jdm4344** [comment]
Hi @azaslonov, is there any ETA available on this issue being resolved?

**Comment by azaslonov** [comment]
@jdm4344, I honestly thought it was fixed. I'll make sure it gets into next release. ETA 3 weeks.

**Comment by jdm4344** [comment]
@azaslonov Great, thanks! And no worries!

**Comment by jdm4344** [comment]
After retesting, issue appears to be resolved. Dropdown can be expanded using keyboard and minimizes when a value is selected. Button element now has necessary accessibility attribute.
